### ISSUE TITLE
Missing files in dataset after upload via CKAN API in parallel

### ISSUE THREAD

**Issue by piodii** [ckan/ckan#5225]
### CKAN Version if known (or site URL)
2.8.3

### Please describe the expected behaviour
All uploaded files should be visible in the dataset.

### Please describe the actual behaviour
The problem relates to the situation when number of files are uploaded in parallel using the CKAN API. In the case when 4 or more files are uploaded in the same time, once whole operation is completed, some of them are not visible in the dataset at the web interface. The API reports ""success": true" status and files are present in the file system. Predominantly, 7 of 10 files are missing in parallel upload (the more files are uploaded the more are missing in percentage). It doesn't happen when the same files are uploaded sequentially.

### What steps can be taken to reproduce the issue? 
Upload multiple files 100MB or larger using the CKAN API in parallel

**Comment by wardi** [comment]
`resource_update` can't safely be used in parallel right because updating metadata is done for the whole package and there's no locking implemented in ckan. Updating multiple resources and uploading multiple files with a single call is supported by the upcoming `package_revise` call, see #4618 

For 2.8 and earlier you will need to serialize your calls to `resource_update` to upload multiple files.

**Comment by piodii** [comment]
I'm not using resource_update. I'm using /api/3/action/resource_create and uploading one file in one call.

**Comment by wardi** [comment]
I should have said `resource_create` or `resource_update`, both update the package metadata so you can't call resource_create in parallel for the same package (dataset)

**Comment by wardi** [comment]
Opened a new ticket for this #6420
